bench/bench_htbl.ml

```diff
@@ -0,0 +1,76 @@
+open Multicore_bench
+
+let run_one ~budgetf ~n_domains ?(n_ops = 20 * Util.iter_factor)
+    ?(n_keys = 10000) ~percent_mem ?(percent_add = (100 - percent_mem + 1) / 2)
+    ?(prepopulate = true) (module Htbl : Htbl_intf.HTBL) =
```

Comment:
Hmm... I suspect that in this particular case it might not matter (I'll try to remember to consider this), but the native OCaml compiler is incapable of specializing (inlining) code when using first-class modules or functors.  What I mean here specifically is that in many cases optimized versions of data structures use a worker-wrapper idiom where the wrapper is designed to be inlined.  Using a first-class module or functor, the compiler will not be able to inline the code at all.  This usually tilts the results such that the optimized data structure will apper relatively slower than it would be under normal use when code would be just directly using the optimized version of the data structure (rather than when using it through a first-class module or functor argument like here) and the compiler is able to inline function calls (when it decides to do so).

IOW, in some cases like _benchmarks_, with the OCaml compiler as it is today, it may be better to just duplicate code, because otherwise one might be tilting the results.

Reply:
Yes, given that both versions use essentially the same code (aside from a short preamble) there is likely not that much difference between the two when it comes to inlining, so in this case using a first-class modules should not tilt the results significantly.

Reply:
I did not know about that. I can still duplicate the code and add a comment so if the code changes at some points, this is not an issue. What do you think ?

Reply:
I don’t think we need to change it in this case, because both hash table implementations should be essentially the same when it comes to inlining, but this is something to keep in mind when writing benchmarks.